### PR TITLE
src/components/global: add FormFieldTextRequired component

### PR DESCRIPTION
**Pull request by maceksimon**
Issue: Testing of individual form fields is bloated when testing the form as a whole. Form fields such as first-name and last-name and job-title share common properties - required text field.

* Add wrapper components `FormFieldTextRequired` to encapsulate the logic of these fields
* Add wrapper components for testing
* Replace fields in forms by the component

### CODE REVIEW

File: src/components/global/FormFieldTextRequired.vue [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      type: String,
+      required: true,
+    },
+    labelShort: {
+      type: String,
+    },
+    autocomplete: {
+      type: String,
+    },
+    testing: {
```
Comment (tmszi) [inline src/components/global/FormFieldTextRequired.vue:55]: `testing` prop is declared, but unused inside component body.

File: src/components/global/FormFieldTextRequired.vue [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+/**
+ * FormFieldTextRequired Component
+ *
+ * The `FormFieldTextRequired` displays text required input.
+ *
+ * @description * Use this component to render text required input in forms.
+ *
+ * Note: This component is commonly used in `FormRegisterCoordinator`, `FormRegisterChallenge`.
+ *
+ * @props
```
Comment (tmszi) [inline src/components/global/FormFieldTextRequired.vue:11]: Missing others component props.

File: src/components/global/FormFieldTextRequired.vue [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+ * - `value` (string, required): The object representing user input.
```
Comment (tmszi) [inline src/components/global/FormFieldTextRequired.vue:12]: `value` prop is valid component prop?

File: src/components/global/FormFieldTextRequired.vue (hunk 1 above)
Comment (maceksimon) [reply src/components/global/FormFieldTextRequired.vue:55]: You are right. I have used it previously to disable `lazy-rules` for easier testing. However, the recent update to Quasar requires that we call .blur() in tests after entering value into the input, so this is no longer needed.
Removed.

File: src/components/global/FormFieldTextRequired.vue (hunk 2 above)
Comment (maceksimon) [reply src/components/global/FormFieldTextRequired.vue:11]: Added other props.

File: src/components/global/FormFieldTextRequired.vue (hunk 3 above)
Comment (maceksimon) [reply src/components/global/FormFieldTextRequired.vue]: No, correct prop is `modelValue`.
Fixed.

**Review by tmszi — APPROVED** [review]

**Pull request merged**